diff --git a/build/legacy/bk-build-utils.js b/build/legacy/bk-build-utils.js
index d7f71fff0..e2af87395 100644
--- a/build/legacy/bk-build-utils.js
+++ b/build/legacy/bk-build-utils.js
@@ -8,6 +8,7 @@
   var fs = require('fs-extra');
   var _ = require('lodash');
   var path = require('path');
+  var log = require('fancy-log');
 
   var prepareBuild = require('./bk-prepare-build');
   var nodePackageFile = require('../package.json');
@@ -101,6 +102,7 @@
         args.push('-ldflags');
         args.push('-X=main.appVersion=' + version);
       }
+      log('Spawning: go "' + args.join('" "') + '", in CWD [' + srcPath + '] with GOPATH: [' + env.GOPATH + ']');
       return spawnProcess('go', args, srcPath, env);
     });
   }
diff --git a/build/legacy/bk-build.js b/build/legacy/bk-build.js
index ceb043f91..d2176c932 100644
--- a/build/legacy/bk-build.js
+++ b/build/legacy/bk-build.js
@@ -236,6 +236,7 @@
   ));
 
   gulp.task('cf-build-backend', gulp.series(
+    'set-no-glide-install', // Doesn't perform a `go build -i` buiild
     'init-build',
     'get-plugins-data',
     'copy-artefacts',
diff --git a/deploy/cloud-foundry/build.sh b/deploy/cloud-foundry/build.sh
index c4add91fc..dceaeb8d8 100755
--- a/deploy/cloud-foundry/build.sh
+++ b/deploy/cloud-foundry/build.sh
@@ -25,6 +25,12 @@ TOP_LEVEL=${CF_DIR}/../../
 
 export STRATOS_TEMP=$(mktemp -d)
 
+function exit_trap() {
+    echo "DEBUG: removing '$STRATOS_TEMP' directory."
+    rm -rf "$STRATOS_TEMP"
+}
+trap exit_trap EXIT
+
 # Copy the config file
 cp ${CF_DIR}/config.properties ${TOP_LEVEL}
 
